src/llama_server_context.cc

```diff
@@ -458,6 +458,15 @@ bool LlamaServerContext::LaunchSlotWithData(LlamaClientSlot*& slot, json data) {
   slot->params.seed = json_value(data, "seed", default_params.seed);
   slot->sparams.grammar = json_value(data, "grammar", default_sparams.grammar);
   slot->sparams.n_probs = json_value(data, "n_probs", default_sparams.n_probs);
+  slot->sparams.min_keep =
+      json_value(data, "min_keep", default_sparams.min_keep);
+  slot->sparams.seed = json_value(data, "seed", default_sparams.seed);
+  slot->sparams.dynatemp_range =
+      json_value(data, "dynatemp_range", default_sparams.dynatemp_range);
+  slot->sparams.dynatemp_exponent =
+      json_value(data, "dynatemp_exponent", default_sparams.dynatemp_exponent);
+  slot->sparams.ignore_eos =
+      json_value(data, "ignore_eos", default_sparams.ignore_eos);
```

Comment:
Can I check my understanding:

- This code filters for top N tokens given sampling settings
- Fills out `completion_probabilities` k-v

Reply:
Actually, the return n probs is in serveral place inside codebase. 
At this part, for every inference step it will add n probs to the result https://github.com/janhq/cortex.llamacpp/blob/main/src/llama_server_context.cc#L1675.

With stream in this line it will form the return json https://github.com/janhq/cortex.llamacpp/blob/main/src/llama_server_context.cc#L919